### PR TITLE
Update to SNIP12 version 1

### PR DESCRIPTION
**Pull request by gaetbout**

### CODE REVIEW

File: scripts/v1/StructWithString.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  return {
+    name: "dappName",
+    version: shortString.encodeShortString("1"),
+    chainId,
+    revision: TypedDataRevision.ACTIVE,
+  };
+}
+
+function getTypedDataHash(myStruct: StructWithString, chainId: string, owner: bigint): string {
+  console.log(JSON.stringify(getTypedData(myStruct, chainId)));
```
Comment (builtbyleo) [inline scripts/v1/StructWithString.ts]: should this be here

**Review by builtbyleo — COMMENTED** [review]
Very nice I think it would be useful to have:
Struct with enum
Struct within struct 
ikd if thats too much work :D

File: scripts/v1/StructWithString.ts (hunk 1 above)
Comment (gaetbout) [reply scripts/v1/StructWithString.ts]: Yeah, it is to be able to copy paste it as a CONST in the `.cairo` file, tbh I should prob print the escaped version 🤔

**Comment by gaetbout** [comment]
struct within struct is u256 and SNIP-12 doesn't support enum, but still maybe there should be an example indeed 🤔

File: scripts/v1/StructWithString.ts (hunk 1 above)
Comment (gaetbout) [reply scripts/v1/StructWithString.ts]: Actually you were right.
Although the idea of printing the escaped version is handy, it is not straigtforward...
Removed them all :D

**Comment by gaetbout** [comment]
Can someone double check about `version: shortString.encodeShortString("1")`.
I think I made a mistake and it shouldn't be encoded. Can someone confirm?

File: src/examples/v1/struct_with_string.cairo [hunk 2]
```diff
@@ -0,0 +1,68 @@
+use starknet::{get_tx_info, get_caller_address};
+use poseidon::PoseidonTrait;
+use hash::{HashStateTrait, HashStateExTrait};
+use off_chain_signature::interfaces::{IOffChainMessageHash, IStructHash, v1::StarknetDomain};
+
+const STRUCT_WITH_U256_TYPE_HASH: felt252 =
+    selector!("\"StructWithString\"(\"some_felt252\":\"felt\",\"some_string\":\"string\")");
```
Comment (sgc-code) [inline src/examples/v1/struct_with_string.cairo]: i would user human readable names to encourage builders to use readable data to sign, For instance "Some String"

Comment (gaetbout) [reply src/examples/v1/struct_with_string.cairo]: Updated all V1 files, as this was not encouraged in V0 I left them as is.
Should I update V0 to use CamelCase naming in the interface for the ts files?
```ts
interface SimpleStruct {
  some_felt252: string;
  some_u128: string;
}
// Will become
interface SimpleStruct {
  someFelt252: string;
  someU128: string;
}
```
?

Comment (sgc-code) [reply src/examples/v1/struct_with_string.cairo]: yeah, in the TS files i'd the js notation "someFelt252". maybe the actual name like "Some Felt 252" but no reason to use the snake_case

**Review by sgc-code — APPROVED** [review]

File: src/examples/v1/struct_with_string.cairo (hunk 2 above)
Comment (gaetbout) [reply src/examples/v1/struct_with_string.cairo]: Still smthng to be done? bit confused as you approved ^^'

Comment (sgc-code) [reply src/examples/v1/struct_with_string.cairo]: ah, no, looks like this was already done

Comment (sgc-code) [reply src/examples/v1/struct_with_string.cairo]: all good to merge

**Pull request merged**